roles/kubernetes/tasks/main.yml

```diff
@@ -1,26 +1,28 @@
 ---
+# Retrieves the kubeconfig for the newly created kubernetes cluster and stores it under $HOME/.kube/<cluster_name>.yml
+- name: Create folder .kube
+  ansible.builtin.file:
+    path: "{{ lookup('env','HOME') }}/.kube/"
+    state: directory
+  when: terraform_run is defined and terraform_run.outputs is defined
+
 - name: Add Helm diff plugin
   kubernetes.core.helm_plugin:
     plugin_path: https://github.com/databus23/helm-diff
     state: present
 # since ionoscloudsdk.ionoscloud version 6, ansible outputs this as changed (in yellow)
+# This gets the Kubeconfig even in check mode so the task is completely disabled when in check mode
 - name: Get k8s config
   ionoscloudsdk.ionoscloud.k8s_config:
     username: "{{ ionos_username }}"
     password: "{{ ionos_password }}"
     k8s_cluster_id: "{{ terraform_run.outputs.cluster_id.value }}"
     config_file: "{{ lookup('env','HOME') }}/.kube/{{ terraform_run.outputs.cluster_name.value }}.yaml"
     state: present
-  when: terraform_run is defined and terraform_run.outputs is defined
+  when: terraform_run is defined and terraform_run.outputs is defined and not ansible_check_mode
```

Comment:
This was necessary because otherwise running the Kubeconfig generation playbook in check mode would result in the admin Kubeconfig being stored locally in ~/.kube